### PR TITLE
respect $fish_prompt_pwd_dir_length

### PR DESCRIPTION
**Pull request by lowne**
... used in default prompt: https://github.com/fish-shell/fish-shell/blob/master/share/functions/prompt_pwd.fish

### CODE REVIEW

**Comment by lowne** [comment]
Good idea; still needs to disable on `0`, so I also added that inline.

File: pwd_info.fish [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
+    command pwd -P | awk -v home="$home" -v git_root="$git_root" -v separator="$separator" -v dir_length="$fish_prompt_pwd_dir_length" '
         function base(string) {
             sub(/^\/?.*\//, "", string)
             return string
         }
         function dirs(string, printLastName,   prefix, path) {
             len = split(string, parts, "/")
             for (i = 1; i < len; i++) {
-                name = substr(parts[i], 1, dir_length)
+                name = dir_length == 0 ? parts[i] : substr(parts[i], 1, dir_length ? dir_length : 1)
```
Comment (jorgebucaran) [inline pwd_info.fish]: @lowne Can you explain this part `dir_length == 0 ? parts[i]`?

Comment (lowne) [reply pwd_info.fish]: in the [default `prompt_pwd`](https://github.com/fish-shell/fish-shell/blob/7e6baab15321a61725e9c78fae2085dcbb9eae7e/share/functions/prompt_pwd.fish#L19), a value of `0` disables shortening.

Comment (jorgebucaran) [reply pwd_info.fish]: Can we simplify that to `name = substr(parts[i], 1, dir_length ? dir_length : 1)` or would that not work?

Comment (lowne) [reply pwd_info.fish]: Wouldn't work. `0` is falsy (at least in the ancient `awk` that comes with macOS), so it'd default back to `1`; and even if it wasn't, `substr(s,n,0)` results in the empty string.

Comment (lowne) [reply pwd_info.fish]: Perhaps I should clarify: this is the behaviour of the default prompt:
```
~> cd path/to/dir
~/p/t/dir> set -xg fish_prompt_pwd_dir_length 2
~/pa/to/dir> set -eg fish_prompt_pwd_dir_length
~/p/t/dir> set -xg fish_prompt_pwd_dir_length 0
~/path/to/dir>
```
The reason for this PR is that I installed a fresh `fish` on a new box, didn't like the shortened PWD so `set -xg fish_prompt_pwd_dir_length 0` in my `config.fish`, later installed (my [fork of](https://github.com/lowne/fish-prompt-metro)) `fish-prompt-metro`, and it reverted back to shortened PWD.

So: `fish_prompt_pwd_dir_length` is now a thing in the default prompt, and although the way it works is perhaps debatable (`0`=disabled, unset=`1`), that's how it is upstream, and I think custom prompt functions should respect the setting.

Comment (jorgebucaran) [reply pwd_info.fish]: Thanks for your time and for the explanation @lowne!

**Pull request merged**